src/data_processing/pid_utils.py

```diff
@@ -0,0 +1,32 @@
+from configparser import SectionProxy
+
+import netCDF4
+import requests
+from requests import HTTPError
+
+
+class PidUtils:
+
+    def __init__(self, config: SectionProxy):
+        self._pid_service_url = config['url']
+
+    def add_pid_to_file(self, filepath: str) -> (str, str):
+        """Queries PID service and adds the PID to NC file metadata."""
+        rootgrp = netCDF4.Dataset(filepath, 'r+')
+        uuid = getattr(rootgrp, 'file_uuid')
+        payload = {
+            'type': 'file',
+            'uuid': uuid
+        }
+        res = requests.post(self._pid_service_url, data=payload)
+
+        try:
+            res.raise_for_status()
+        except HTTPError:
+            raise HTTPError(f'PID service failed with status {res.status_code}:\n{res.json()["detail"]}')
+
+        pid = res.json()['pid']
+        rootgrp.pid = pid
+        rootgrp.close()
+
+        return uuid, pid
```

Comment:
Tarvitseeko olla luokka? Jos luokkaan ei ole tulossa muita metodeja, niin `add_pid_to_file` voisi olla vain funktio. Muutenkin koko file on aika tynkä, joten jos tänne ei ole tulossa mitään muuta niin voisi harkita, että sen funktion siirtäisi johonkin toiseen moduuliin.

Reply:
Alun perin laitoin `add_pid_to_file`-funktion `utils`in alle, mutta ongelmaksi koitui se että tuo funktio tarvitsee configista pid-palvelun URL:n. Se olisi pitänyt passata process-cloudnetissa ympäriinsä, jos funktio ei ole luokassa, joka initialisoidaan siellä missä config määritelty.